tests/actions/PolicyTagTest.ts

```diff
@@ -307,95 +308,129 @@ describe('actions/Policy', () => {
-                                    expect(newTag?.errors).toBeFalsy();
-                                    expect(newTag?.pendingAction).toBe(CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
+            // Verify optimistic data
+            const policyTagsOptimistic = await OnyxUtils.get(`${ONYXKEYS.COLLECTION.POLICY_TAGS}${fakePolicy.id}`);
+            const newTagOptimistic = policyTagsOptimistic?.[tagListName]?.tags?.[newTagName];
+            expect(newTagOptimistic?.name).toBe(newTagName);
+            expect(newTagOptimistic?.enabled).toBe(true);
+            expect(newTagOptimistic?.errors).toBeFalsy();
+            expect(newTagOptimistic?.pendingAction).toBe(CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
+
+            mockFetch?.resume?.();
+            await waitForBatchedUpdates();
 
-                                    resolve();
-                                },
-                            });
-                        }),
-                )
-                .then(mockFetch?.resume)
-                .then(waitForBatchedUpdates)
-                .then(
-                    () =>
-                        new Promise<void>((resolve) => {
-                            const connection = Onyx.connect({
-                                key: `${ONYXKEYS.COLLECTION.POLICY_TAGS}${fakePolicy.id}`,
-                                waitForCollectionCallback: false,
-                                callback: (policyTags) => {
-                                    Onyx.disconnect(connection);
+            // Verify success data
+            const policyTagsSuccess = await OnyxUtils.get(`${ONYXKEYS.COLLECTION.POLICY_TAGS}${fakePolicy.id}`);
+            const newTagSuccess = policyTagsSuccess?.[tagListName]?.tags?.[newTagName];
+            expect(newTagSuccess?.errors).toBeFalsy();
+            expect(newTagSuccess?.pendingAction).toBeFalsy();
+        });
 
-                                    const newTag = policyTags?.[tagListName]?.tags?.[newTagName];
-                                    expect(newTag?.errors).toBeFalsy();
-                                    expect(newTag?.pendingAction).toBeFalsy();
+        it('reset new policy tag when api returns error', async () => {
+            const fakePolicy = createRandomPolicy(0);
+            fakePolicy.areTagsEnabled = true;
 
-                                    resolve();
-                                },
-                            });
-                        }),
-                );
+            const tagListName = 'Fake tag';
+            const newTagName = 'new tag';
+            const fakePolicyTags = createRandomPolicyTags(tagListName);
+
+            mockFetch?.pause?.();
+
+            await waitForBatchedUpdates();
+            mockFetch?.fail?.();
+            createPolicyTag(fakePolicy.id, newTagName, fakePolicyTags);
+            await waitForBatchedUpdates();
+
+            mockFetch?.resume?.();
+            await waitForBatchedUpdates();
+
+            // Verify error state
```

Comment:
NAB: It's really helpful in tests to use the Given/When/Then format as it provides more direct context into what the test is doing and what exactly is being verified.

You can read all about how to do this here: https://github.com/Expensify/App/blob/2e6c26631dc3e8ade7250537ffe49cc0a18e3ad5/tests/README.md#documenting-tests